tests/components/test_kubernetes_app.py

```diff
@@ -69,6 +69,7 @@ def test_should_lazy_load_helm_wrapper_and_not_repo_add(
         mocker: MockerFixture,
         helm_mock: MagicMock,
         app_value: KubernetesTestValue,
+        dry_run_handler: MagicMock,
```

Comment:
It is needed in this case because the dry-run is set to try when calling the `deploy`, and the dry-run handler is executed. When passing the MagicMock, the actual execution is skipped. I can add a validation `assert_called_once`.
Another way would be to remove it and call deploy with dry-run Flase.

Reply:
IMO the correct way would be to split this into two tests. Once with dry_run true and once with false. But this is outside of the scope of this PR. So maybe setting it to false is the better approach for now

Reply:
Changed